core/interpreter_js/src/core_to_map_bindings/mod.rs

```diff
@@ -110,11 +111,11 @@ fn traverse_object<'ctx>(
 
 /// Enum for formatting JsValues in a format suitable for debugging.
 enum JSValueDebug<'a> {
-    Ref(JSValueRef<'a>),
+    Ref(&'a JSValueRef<'a>),
```

Comment:
This is probably a dumb question, but I have to ask to understand. Why does it change from owned value to reference? Or it isn't directly related to the fix, and it just improves original code?

Reply:
So, after opening changes in Javy, I think I understand why. You started to work with reference of JSValueRef in encode/decode buffer.

Reply:
Yeah. JSValueRef can be imagined as `Rc<JSValueRaw>`, but the Rc is implemented in QuickJS itself. In some places in QuickJS which take in a `JSValueRaw` it takes it by move semantics (it's always a handle, but it acts as if it owned the value after and frees it) and some places take it as a reference.

I tried to reflect this in the Javy codebase as well to make it clear that the value is reference counted, sometimes moved instead of just referenced and has a non-trivial drop (it might trigger a GC).